cpu/lpc23xx/periph/timer.c

```diff
@@ -167,6 +168,37 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
     dev->MR[channel] = value;
     /* Match Interrupt */
     dev->MCR |= (1 << (channel * 3));
+
+    return 0;
+}
+
+int timer_set(tim_t tim, int channel, unsigned int timeout)
+{
+    if (((unsigned) tim >= TIMER_NUMOF) || ((unsigned) channel >= TIMER_CHANNEL_NUMOF)) {
+        return -1;
+    }
+
+    lpc23xx_timer_t *dev = get_dev(tim);
+
+    unsigned state = irq_disable();
+    unsigned absolute = timeout + dev->TC;
+    unsigned mask = 1 << (channel * 3);
+
+    dev->MR[channel] = absolute;
+    dev->MCR |= mask;
+    set_oneshot(tim, channel);
+
+    if (absolute - dev->TC > timeout) {
+        /* Timer already expired. Trigger the interrupt now and loop until it
+         * is triggered.
+         */
+        while ((dev->IR & mask) == 0) {
+            dev->MR[channel] = dev->TC;
+        }
```

Comment:
I tested with:

``` C
        while ((dev->IR & mask) == 0) {
            dev->MR[channel] = dev->TC;
            dev->MCR |= mask;
        }
```

and even with

``` C
        if ((dev->IR & mask) == 0) {
            dev->MR[channel] = dev->TC + 2;
            dev->MCR |= mask;
        }
```

All to no avail. It is a pitty that we cannot do

``` C
        dev->IR |= mask;
```

to just set the interrupt from software :-/

Reply:
If we stop the timer while configuring the alarm it works down to "interval 1".

Is "interval 0" even a sensible thing to configure a timer to?
We could do

```C
if (timeout == 0) {
    return -1;
}
```

or 

```C
if (timeout == 0) {
    timeout = 1;
}
```

Reply:
To me, `timeout == 0` means the interrupt should fire as soon as possible. If this means using `timeout ? timeout : 1`, this would be fine to me.

Reply:
> If we stop the timer while configuring the alarm it works down to "interval 1".

I'm not sure if implicitly stopping the timer is a good idea. This would result in reduced accuracy in time keeping. I wouldn't expect a clock to run slower when frequently setting a timeout.

Reply:
Ok